packages/types/src/types/requests/datatrak-web-server/SurveyRequest.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import {
+  Question,
+  Survey,
+  SurveyScreen as BaseSurveyScreen,
+  SurveyScreenComponent as BaseSurveyScreenComponent,
+} from '../../models';
+import { KeysToCamelCase } from '../../../utils/casing';
+
+export type Params = Record<string, never>;
+
+type VisibilityCriteria = Record<Question['id'], any> & {
+  _conjunction?: string;
+};
+
+type ValidationCriteria = {
+  required?: boolean;
+  min?: number;
+  max?: number;
+};
+
+// Separating these out because sometimes the camel casing of Record<string, unknown> is not then identitied as still being a Record<string, unknown>
+
+type CamelCasedQuestion = KeysToCamelCase<Omit<Question, 'options'>>;
+
+type CamelCasedComponent = KeysToCamelCase<
+  Omit<
+    BaseSurveyScreenComponent,
+    'validation_criteria' | 'visibility_criteria' | 'config' | 'id' | 'question_label' | 'screen_id'
+  >
+>;
+
+export type CodeGeneratorConfig = {
+  type: 'shortid' | 'mongoid';
+  prefix?: string;
+  length?: number;
+  chunkLength?: number;
+  alphabet?: string;
+};
+
+export type AutocompleteConfig = {
+  createNew?: boolean;
+  attributes?: {
+    [key: string]: { questionId: Question['id'] };
+  };
+};
+
+type Config = {
+  codeGenerator?: CodeGeneratorConfig;
+  autocomplete?: AutocompleteConfig;
```

Comment:
It would be great to add the entity config here too which I currently have in the survey submit types but I'm happy to do that after we have merged our two PRs together.